src/main/webapp/app/guided-tour/guided-tour.component.ts

```diff
@@ -429,14 +455,35 @@ export class GuidedTourComponent implements AfterViewInit, OnDestroy {
     /**
      * Update tour step location and return selected element as DOMRect
      * @param selectedElement: selected element in DOM
-     * @return {selectedElementRect} selected element as DOMRect or null
+     * @return selected element as DOMRect or null
      */
     public updateStepLocation(selectedElement: HTMLElement | null): DOMRect | null {
         let selectedElementRect = null;
 
         if (selectedElement) {
             selectedElementRect = selectedElement.getBoundingClientRect() as DOMRect;
+            if (this.currentTourStep && this.currentTourStep.enableUserInteraction) {
+                this.clickEventListener = () => this.handleUserClickInteraction(selectedElement);
+                selectedElement.addEventListener('click', this.clickEventListener);
+            }
         }
+
         return selectedElementRect;
     }
+
+    /* ==========     User interaction methods     ========== */
+
+    /**
+     * Handles the user click interaction with the highlighted element, and skips tour if the interaction
+     * is not intended
+     */
+    public handleUserClickInteraction(selectedElement: HTMLElement) {
+        selectedElement.removeEventListener('click', this.clickEventListener);
+        if (!this.currentTourStep) {
+            return;
+        }
+        setTimeout(() => {
```

Comment:
It's not bad by any means but you can feel the lag when clicking on the element until the tour step moves to the next step. The transition is not smooth because the tour stays at the old element, waits for the new element to appear and then moves.
This could e.g. be improved by a loading indicator and/or removing the old step earlier.